nncf/onnx/graph/nncf_graph_builder.py

```diff
@@ -21,11 +21,198 @@
+    If node has bias tensor, information for algorithms about bias is stored in bias_attrs.
+    If node has attibutes needed for algorithms, they are stored in node_attrs.
+    E.g. 'transA' attirbute of Gemm node for Quantization.
+    """
+
+    def __init__(
+        self,
+        weight_attrs: Optional[Dict[int, Dict]] = None,
+        bias_attrs: Optional[Dict[str, Any]] = None,
+        node_attrs: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        :param weight_attrs: Maps input port id asocciated with weight to a weight description.
+        :param bias_attrs: Maps bias tensor name asocciated with weight to a weight description.
+        :param node_attrs: Maps attribute name to an attribute value.
+        """
+        self.weight_attrs = weight_attrs if weight_attrs is not None else {}
+        self.bias_attrs = bias_attrs if bias_attrs is not None else {}
+        self.node_attrs = node_attrs if node_attrs is not None else {}
+
+    def has_weight(self) -> bool:
+        return bool(self.weight_attrs)
+
+    def has_bias(self) -> bool:
+        return bool(self.bias_attrs)
+
+    def has_node_attrs(self) -> bool:
+        return bool(self.node_attrs)
+
+
+def _get_weight_edge_name(onnx_graph: ONNXGraph, node: onnx.NodeProto, port_id: int) -> Optional[str]:
+    """
+    Returns an edge name associated with a weight of a node laying on  an input port_id.
+
+    Checks whether a node has a tensor on input port_id.
+    If does then it is a weight and returns corresponding edge name.
+    If not - take a parent node into this port id and does the same check for it.
+
+    If an edge with a weight was not found then returns None.
+
+    METATYPES THAT COULD CONSUME A WEIGHT TENSOR:
+        ONNXConstantMetatype
+        ONNXIdentityMetatype
+        ONNXReshapeMetatype
+        ONNXTransposeMetatype
+        ONNXQuantizeLinearMetatype
+
+    :param onnx_graph: ONNXGraph.
+    :param node: Node.
+    :param port_id: Port id on which a weight edge is seeking.
+    :return: Edge name associated with a weight.
+    """
+    WEIGHT_CONSUMING_TYPES = (
+        ONNXConstantMetatype.get_all_aliases()
+        + ONNXIdentityMetatype.get_all_aliases()
+        + ONNXReshapeMetatype.get_all_aliases()
+        + ONNXTransposeMetatype.get_all_aliases()
+        + ONNXQuantizeLinearMetatype.get_all_aliases()
+    )
+    PROPAGATING_ONLY_TYPES = ONNXDequantizeLinearMetatype.get_all_aliases()
+
+    if node.op_type not in PROPAGATING_ONLY_TYPES and onnx_graph.has_tensor(node.input[port_id]):
+        if node.op_type in ONNXReshapeMetatype.get_all_aliases():
+            # Only Reshape nodes returns output edge.
+            return node.output[0]
+        return node.input[port_id]
+
+    parent = onnx_graph.get_parent(node, port_id)
+    if parent and parent.op_type in (WEIGHT_CONSUMING_TYPES + PROPAGATING_ONLY_TYPES):
+        return _get_weight_edge_name(onnx_graph, parent, 0)
```

Comment:
This function returns a name of a tensor which is a node's weight. 
There are nodes that can consume weight tensor on 0-input port:
```
        ONNXConstantMetatype
        ONNXIdentityMetatype
        ONNXReshapeMetatype
        ONNXTransposeMetatype
        ONNXQuantizeLinearMetatype
```
There are two actions:
1) Getting weight tensor
2) Propagating

There are two node types:
1) Node that can have weight tensor and can be propagated further.
2) Node that is only for propagation.

If the node in the type 1, then we try to get weight tensor from it. If there is no tensor, we take a parent on that input port and if the type lays in these two, do the same for this node.
If the node in the type 2, we take a parent on that input port and if the type lays in these two, do the same for this node.

> Also, It's not clear, why we always walk through 0-port.

Because the nodes that consume weight tensor they must consume it only on 0 input port